modify OAuth IMAP (#80)
pluginsGLPI/doc: source/oauthimap/index.rst

OAuth IMAP
==========

.. note::

>    Microsoft is gradually removing the TLS 1.0 and 1.1 protocols for all Microsoft 365 applications. In order to keep your collector running, you need to add an **Oauth IMAP** plugin available in the marketplace.

Reviewer: ```suggestion
   Microsoft is gradually removing the TLS 1.0 and 1.1 protocols for all Microsoft 365 applications. In order to keep your collector running, you need to add the **Oauth IMAP** plugin, which is available in the marketplace.
```

.. tip::

>    The Oauth tokens for the collector, retrieved during authentication with Azure by the OauthIMAP plugin, are “offline” tokens that delegate authorisation to a third-party application (GLPI). These tokens provide a renewal code that will be used by the application to renew them automatically without user intervention. You will therefore not be asked to re-authenticate after the 1st authorisation request (unless you change your password at a later date).

Reviewer 2: > (unless you change your password at a later date)

Are you sure of that? OAuth tokens should not be impacted by a password change.

Source and Download
-------------------

* Sources link: https://github.com/pluginsGLPI/oauthimap
* Download: https://github.com/pluginsGLPI/oauthimap/releases

Requirements (on-premise)
-------------------------

============ =========== ===========
GLPI Version Minimum PHP Recommended
============ =========== ===========
10.0.x       8.1         8.2
============ =========== ===========

Here are the configuration steps including configuration phases on the Entra side.

Install the plugin
------------------

.. note::

>    This plugin is community and does not need a license. It is also available in Cloud.

Reviewer: ```suggestion
   This plugin does not require a GLPI-Network subscription. It is also available in Cloud.
```

> -  See you in the marketplace. Download Oauth IMAP and enable it

Reviewer: ```suggestion
-  Go to the marketplace. Download Oauth IMAP and enable it
```

.. figure:: images/oauth-imap-1.png
   :alt:

> -  Meet now on your `Azure tenant <https://portal.azure.com/#home>`__
> -  In the search box type **registration** then select
>    **app registrations**

Reviewer: ```suggestion
-  Open the `Azure Portal <https://portal.azure.com/#home>`__ for your tenant
-  In the search box type **registration** then select
   **App registrations**
```

.. figure:: images/oauth-imap-2.png
   :alt:

Register your Entra application
-------------------------------

Create the application
~~~~~~~~~~~~~~~~~~~~~~

> -  Click on **new registration**
> -  Indicate the desired name, select the type of account supported then indicate the redirection URL (present in the configuration of the plugin from your GLPI interface:
>    https://XXXXXXXXXXXXXX/marketplace/oauthimap/front/authorization.callback.php) specifying the **Web** option
> -  Then click on **register**.

Reviewer: ```suggestion
-  Click on **New registration**
-  Enter the desired name, select the type of account supported then enter the redirection URL (present in the configuration of the plugin from your GLPI interface:
   https://XXXXXXXXXXXXXX/marketplace/oauthimap/front/authorization.callback.php) specifying the **Web** option
-  Then click on **Register**.
```

.. figure:: images/oauth-imap-3.png
   :alt:

Add a secret
~~~~~~~~~~~~

-  In the **Certificates and secrets** tab
-  Click on **Client secrets**
-  Then **New client secret**

.. figure:: images/oauth-imap-4.png
   :alt:

-  Enter a description and then an expiration date
-  A secret **value** is then generated. Keep this value well because once you have left this page, it will no longer be recoverable

.. figure:: images/oauth-imap-5.png
   :alt:

-  Return to the **Overview** tab and **copy** the following values ​​and the secret seen above

.. figure:: images/oauth-imap-6.png
   :alt:

setup GLPI
----------

-  Now go back to your GLPI interface **configuration > Application Aouth IMAP** and indicate the information collected previously :

.. figure:: images/oauth-imap-7.png
   :alt:

-  Click **add**

> -  Now in the **Oauth authorization** tab , we will **create an authorization**

Reviewer: ```suggestion
-  Now in the **Oauth authorization** tab, click **Create an authorization**
```

.. figure:: images/oauth-imap-8.png
   :alt:

> -  When you click on :ti-add:**create authorization**, you will be redirected to the Microsoft services sign-in page

Reviewer: ```suggestion
-  When you click on :ti-add:**Create authorization**, you will be redirected to the Microsoft services sign-in page
```
